AWSTranscribeStreaming/Internal/AWSTranscribeStreamingWebSocketProvider.h

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol AWSTranscribeStreamingWebSocketProvider <NSObject>
+
+@property(nonatomic, strong) id<AWSTranscribeStreamingClientDelegate> clientDelegate;
+@property(nonatomic, strong) id<AWSTranscribeStreamingWebSocketProvider> webSocket;
```

Comment:
Sorry I just clued into this. This property seems redundant--why does an object conforming to `AWSTranscribeStreamingWebSocketProvider` have a property of `AWSTranscribeStreamingWebSocketProvider`, which presumably returns `self`?

Reply:
ha this caused me issues in my unit tests and I honestly couldnt remember why it was here. probably left over from one of the refactors so I left it and worked around it but you're right it shoudl be removed.